File: usaco/section1/beads.py
'''
TASK: beads
LANG: PYTHON3
ID: futzipe1
'''
import math
data = open('beads.in', 'r')
# index, s = 
data = data.read().split('\n')
index = data[0]
s = data[1]
s = s.strip()
index = int(index)



def printCurrentIndex(index):
  print(coolString)
  print(" "*index + '^')

def findValues(up, index):
    value = 1
    iteration = 0
    if (up == -1):
      initValue = coolString[index-1]
      index -= 1
    initValue = coolString[index]
    tIndex = index
    while ( ((tIndex < len(coolString) - 1) and tIndex > 0) and initValue == 'w'):
        initValue = coolString[tIndex + up]
        tIndex += up
    while ( index < len(coolString) and index > 0) and (coolString[index] == initValue or coolString[index] == 'w'):
        value += 1
        index += up
        iteration += 1
 
    return value


maxe = 0
coolString = s * 3
initValue = coolString[index]
for ite in range(index, 2 * index):
    print(ite)
    lVal = findValues(1, ite)
    rVal = findValues(-1, ite)
    cVal = lVal + rVal -2
    if (cVal > maxe):
        maxe = cVal

f = open("beads.out", "w")

if (maxe > index):
    f.write(str(index)+"\n")
    f.close()
else:
    f.write(str(maxe)+"\n")
    f.close()

File: usaco/section1/beads.cpp
/*
ID: futzipe1
TASK: beads
LANG: C++                 
*/
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

ofstream fout("beads.out");
ifstream fin("beads.in");
int strL;
string s;
int maxe;
int index;
string coolString;

int findValues(int up, int index)
{
    int value = 1, iteration = 0;
    char initValue = coolString[index];
    cout << "initValue: " << initValue << " ";
    int tIndex = index;
    {
        while (initValue == 'w')
        {
            initValue = coolString[tIndex + up];
            tIndex += up;
        }
        while (coolString[index + up] == initValue || coolString[index + up] == 'w' && iteration < strL / 2)
        {
            cout << coolString[index + up];
            value++;
            index += up;
            iteration++;
        }
        cout << endl;
        return value;
    }
}

int main()
{
    fin >> strL >> s;
    maxe = 0;
    index = strL;
    coolString = s + s + s;
    char initValue = s[index];
    bool special = false;
    int iter = 1;
    // while (iter < strL)
    // {
    //     if (s[iter] == initValue && iter == strL - 1)
    //     {
    //         special = true;
    //         break;
    //     }
    //     iter++;
    // }
    // if (special)
    // {
    //     fout << strL << endl;
    //     return 0;
    // }
    for (; index < 2 * strL; index++)
    {
        int lValue = findValues(1, index);
        cout
            << index << "lvalue: " << lValue << endl;
        int rValue = findValues(-1, index);
        cout << endl
             << "rValue: " << rValue << endl;
        int cValue = lValue + rValue;
        if (cValue > maxe)
            maxe = cValue;
    }
    if (maxe > strL)
        fout << strL << endl;
    else
        fout << maxe << endl;
    return 0;
}
